### PR TITLE
EDGECLOUD-6088 Demo app - Add support for QosPrioritySession APIs

### PR DESCRIPTION
**Pull request by brucearmstrong**
- Removed direct calls to DT's REST API server.
- "Update QOS Session" dialog now calls SDK's QosPrioritySession create/delete methods via MatchingEngineHelper.java.
- Removed getLocalIp methods and now call SDK's public methods.
- Removed MEL vestiges.

### CODE REVIEW

File: android/MobiledgeXSDKDemo/app/src/main/java/com/mobiledgex/sdkdemo/DtQosPrioritySessions.java [hunk 1]
```diff
@@ -46,10 +49,11 @@ protected static void createPrioritySession(MainActivity activity, MatchingEngin
         Log.i(TAG, "meHelper.mQosSessionId="+meHelper.mQosSessionId+" meHelper.mQosProfileName="+meHelper.mQosProfileName);
         // Create dialog with QOS profile names.
         List<String> items = new ArrayList<>();
-        items.add("LOW_LATENCY");
-        items.add("THROUGHPUT_S");
-        items.add("THROUGHPUT_M");
-        items.add("THROUGHPUT_L");
+        for (AppClient.QosSessionProfile profileName : AppClient.QosSessionProfile.values()) {
```
Comment (lgarner) [inline android/MobiledgeXSDKDemo/app/src/main/java/com/mobiledgex/sdkdemo/DtQosPrioritySessions.java:52]: Why name(), and not the enum const compare?

File: android/MobiledgeXSDKDemo/matchingenginehelper/src/main/java/com/mobiledgex/matchingenginehelper/MatchingEngineHelper.java [hunk 2]
```diff
@@ -492,11 +491,140 @@ public void registerClientInBackground() {
...
+        builder.setProtocolIn(AppClient.QosSessionProtocol.valueOf(mAppInstProto));
+        builder.setProfile(AppClient.QosSessionProfile.valueOf(qosProfile));
+        builder.setSessionDuration(duration);
+
+        AppClient.QosPrioritySessionCreateRequest qosPrioritySessionCreateRequest;
+        Future<AppClient.QosPrioritySessionReply> qosPrioritySessionReplyFuture;
+        qosPrioritySessionCreateRequest = builder.build();
+        qosPrioritySessionReplyFuture = me.qosPrioritySessionCreateFuture(qosPrioritySessionCreateRequest, mDmeHostname, mDmePort,10000);
+        Log.i(TAG, "qosPrioritySessionReplyFuture="+qosPrioritySessionReplyFuture);
+        mQosPrioritySessionReply = qosPrioritySessionReplyFuture.get();
```
Comment (lgarner) [inline android/MobiledgeXSDKDemo/matchingenginehelper/src/main/java/com/mobiledgex/matchingenginehelper/MatchingEngineHelper.java:516]: This retrieves a future, then call .get(), defeating reason for Future along async paths. A completableFuture is more chain able?

File: android/MobiledgeXSDKDemo/matchingenginehelper/src/main/java/com/mobiledgex/matchingenginehelper/MatchingEngineHelper.java [hunk 3]
```diff
@@ -492,11 +491,140 @@ public void registerClientInBackground() {
...
+        Log.i(TAG, "mQosPrioritySessionReply.getSessionId()="+ mQosPrioritySessionReply.getSessionId()+" mQosPrioritySessionReply.getHttpStatus()="+ mQosPrioritySessionReply.getHttpStatus());
+        if (mQosPrioritySessionReply.getSessionId() == null || mQosPrioritySessionReply.getSessionId().length() == 0) {
```
Comment (ah1053) [inline android/MobiledgeXSDKDemo/matchingenginehelper/src/main/java/com/mobiledgex/matchingenginehelper/MatchingEngineHelper.java:518]: thats is actually better than what I did with C# just check for session ID then log the httpStatus if the request failed.

File: android/MobiledgeXSDKDemo/app/src/main/java/com/mobiledgex/sdkdemo/DtQosPrioritySessions.java (hunk 1 above)
Comment (brucearmstrong) [reply android/MobiledgeXSDKDemo/app/src/main/java/com/mobiledgex/sdkdemo/DtQosPrioritySessions.java:52]: Will fix.

File: android/MobiledgeXSDKDemo/matchingenginehelper/src/main/java/com/mobiledgex/matchingenginehelper/MatchingEngineHelper.java [hunk 4]
```diff
@@ -492,11 +491,140 @@ public void registerClientInBackground() {
...
+        return true;
+    }
+
+    public void qosPrioritySessionCreateInBackground(String qosProfile, int duration) {
+        new Thread(() -> {
+            try {
+                qosPrioritySessionCreate(qosProfile, duration);
+            } catch (ExecutionException | InterruptedException
+                    | PackageManager.NameNotFoundException | IllegalArgumentException e) {
+                Log.e(TAG, "Exception in registerClientInBackground() for "+
```
Comment (ah1053) [inline android/MobiledgeXSDKDemo/matchingenginehelper/src/main/java/com/mobiledgex/matchingenginehelper/MatchingEngineHelper.java:538]: registerClientInBackground >  qosPrioritySessionCreateInBackground

Comment (brucearmstrong) [reply android/MobiledgeXSDKDemo/matchingenginehelper/src/main/java/com/mobiledgex/matchingenginehelper/MatchingEngineHelper.java:538]: @ah1053 Can you elaborate on this comment?

Comment (ah1053) [reply android/MobiledgeXSDKDemo/matchingenginehelper/src/main/java/com/mobiledgex/matchingenginehelper/MatchingEngineHelper.java:536]: The Log text is for register client, Do you you think this try catch block should refer to qos instead

Comment (ah1053) [reply android/MobiledgeXSDKDemo/matchingenginehelper/src/main/java/com/mobiledgex/matchingenginehelper/MatchingEngineHelper.java:536]: "Exception in registerClientInBackground() for " > is that suppose to be "Exception in qosPrioritySessionCreateInBackground() for "

**Pull request merged**